### DISCUSSION TITLE
NVS("label").commit()    doubt

### DISCUSSION THREAD

**Discussion by nastiliano** [micropython/micropython#12756] [General]
Hello everyone.

I read the documentation about ESP32 and NVS (Non-volatile Storage) and it says that:

"Changes to NVS need to be committed to flash by calling the commit method. Failure to call commit results in changes being lost at the next reset."

But my doubt is if I must use "commit()" after every ".set" operation or I can define:

    NVS_page = esp32.NVS("NVS_page_label")

And make, for example, 10 write operations and finally close with a "commit" operation before reset the device.

    ...set:blob()
    ...set_i32()
    ...

Thank you very much.

**Comment by shariltumin** [comment]
From some tests I did, it seems that the ```nvs.commit()``` is not needed, new blobs are written and keep their values even after ```machine.reset()``` and power off/on reset.

However, it is safer to do a ```nvs.commit()```, but not necessarily after every set.

```py
MicroPython v1.22.0-preview.1.g523284777.dirty on 2023-10-06; ESP32 DEVK1 with ESP32
>>> import esp32
>>> nvsa = esp32.NVS('A')
D (22878) nvs: nvs_open_from_partition A 1
>>> buf = bytearray(20)
>>> nvsa.set_blob('a', 'The key was an a')
D (161578) nvs: nvs_set_blob a 16
>>> nvsa.set_blob('b', 'The next key is a b')
D (187208) nvs: nvs_set_blob b 19
>>> nvsa.get_blob('a',buf)
D (197768) nvs: nvs_get_str_or_blob a
16
>>> buf
bytearray(b'The key was an a\x00\x00\x00\x00')
>>> nvsa.get_blob('b',buf)
D (211618) nvs: nvs_get_str_or_blob b
19
>>> buf
bytearray(b'The next key is a b\x00')
>>> from machine import reset
>>> reset()

..... machine.reset() ......

>>> import esp32
>>> nvsa = esp32.NVS('A')
D (46288) nvs: nvs_open_from_partition A 1
>>> buf = bytearray(20)
>>> nvsa.get_blob('a',buf)
D (60388) nvs: nvs_get_str_or_blob a
16
>>> nvsa.get_blob('a',buf)
D (85758) nvs: nvs_get_str_or_blob a
16
>>> buf
bytearray(b'The key was an a\x00\x00\x00\x00')
>>> nvsa.get_blob('b',buf)
D (121888) nvs: nvs_get_str_or_blob b
19
>>> buf
bytearray(b'The next key is a b\x00')
>>> nvsb = esp32.NVS('B')
D (663308) nvs: nvs_open_from_partition B 1
>>> nvsb.set_blob('a', 'Hello there!')
D (734618) nvs: nvs_set_blob a 12

.... POWER OFF ....

>>> import esp32
>>> nvsa = esp32.NVS('A')
D (20388) nvs: nvs_open_from_partition A 1
>>> nvsb = esp32.NVS('B')
D (26038) nvs: nvs_open_from_partition B 1
>>> buf = bytearray(20)
>>> nvsa.get_blob('a',buf)
D (80728) nvs: nvs_get_str_or_blob a
16
>>> buf
bytearray(b'The key was an a\x00\x00\x00\x00')
>>> nvsa.get_blob('b',buf)
D (95198) nvs: nvs_get_str_or_blob b
19
>>> buf
bytearray(b'The next key is a b\x00')
>>> nvsb.get_blob('a',buf)
D (118068) nvs: nvs_get_str_or_blob a
12
>>> buf
bytearray(b'Hello there! is a b\x00')
>>>
```

**Reply by nastiliano** [reply]
Ok, thank you.

I suposed this.
I used to put a ".commit()" operation after aprox. 10 ".set()" operations before resetting device, but now knowing that, I will put only one ".commit()" after all my ".set()" operations before reseting device to asure the correct data saving.


Thank you, best regards.
